### PR TITLE
donor defnitions

### LINKED ISSUE #466: Fill in more definitions for Donor metadata terms
Should have something for all of them, not just the obscure ones.
(Work is actually in ingest-validation-tools: Defs get pulled in with submodule.)

### PR DESCRIPTION
**Pull request by mccalluc**
Fix #466. The portal-docs are not actually helpful here: Jonathan is describing the EAV structure, not giving definitions for the attributes he uses.

### CODE REVIEW

**Comment by cebriggs7135** [comment]
@mccalluc What are EAV structures?

**Comment by mccalluc** [comment]
@cebriggs7135 -- EAV = "Entity Attribute Value" 

If you look at https://portal.hubmapconsortium.org/docs/donor you can see that what Jonathan is describing is more about the structure of the donor metadata, and not really about the particular fields it contains.

File: src/ingest_validation_tools/table-schemas/donor.yaml [hunk 1]
```diff
@@ -2,31 +2,34 @@
 # The full description of Donor metadata is at
 # https://portal.hubmapconsortium.org/docs/donor
 
+# Most definitions taken from:
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/donor.yaml:5]:
```suggestion
# Some definitions taken from:

```

File: docs/field-descriptions.yaml [hunk 2]
```diff
@@ -219,6 +239,8 @@ sequencing_read_format: Slash-delimited list of the number of sequencing cycles
   for example, Read1, i7 index, i5 index, and Read2.
 sequencing_read_percent_q30: Percent of bases with Quality scores above Q30
 sequencing_reagent_kit: Reagent kit used for sequencing
+sex: Phenotypic expression of chromosomal makeup that defines a study subject as male,
+  female, or other. Compare to gender.
```
Comment (cebriggs7135) [inline docs/field-descriptions.yaml:243]:
```suggestion
sex: Biological sex at birth: male or female or other.
```

File: src/ingest_validation_tools/table-schemas/donor.yaml [hunk 3]
```diff
@@ -2,31 +2,34 @@
...
-#   description: TODO
-# -
-#   name: blood_type
-#   description: TODO
+-
+  name: age_value
+  description:  The time elapsed since birth. 
+-
+  name: blood_type
+  description:  'The specific reaction pattern of erythrocytes of an individual to the antisera of one blood group; e.g., the ABO blood group consists of four major blood types: O, A, B, and AB. This classification depends on the presence or absence of two major antigens: A or B. Type O occurs when neither is present and type AB when both are present. The blood type is the genetic phenotype of the individual for one blood group system and may be determined using different antisera available for testing.'
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/donor.yaml:17]:
```suggestion
  description: ABO blood type or "serotype" refers to the presence/absence of the either/both A & B blood antigens.
```

File: src/ingest_validation_tools/table-schemas/donor.yaml [hunk 4]
```diff
@@ -2,31 +2,34 @@
...
+  description:  The circumstance or condition that results in the death of a living being.
 # -
 #   name: death_event
 #   description: TODO
-# -
-#   name: height_unit
-#   description: TODO
+-
+  name: height_unit
+  description:  The vertical measurement or distance from the base to the top of a subject or participant.
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/donor.yaml:32]:
```suggestion
  description:   The unit of measurement of the body length of a subject from feet to the head.

```

File: src/ingest_validation_tools/table-schemas/donor.yaml [hunk 5]
```diff
@@ -2,31 +2,34 @@
...
 #   description: TODO
-# -
-#   name: height_unit
-#   description: TODO
+-
+  name: height_unit
+  description:  The vertical measurement or distance from the base to the top of a subject or participant. 
 # -
 #   name: height_value
 #   description: TODO
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/donor.yaml:35]:
```suggestion
#   description:  The  body length of a subject from feet to the head.

```

File: src/ingest_validation_tools/table-schemas/donor.yaml [hunk 6]
```diff
@@ -49,21 +52,22 @@ fields:
...
-# -
-#   name: race
-#   description: TODO
-# -
-#   name: sex
-#   description: TODO
+-
+  name: mechanism_of_injury
+  # From https://iaedjournal.org/
+  description: How, with what force, and on which part of the body the patient was injured.
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/donor.yaml:58]:
```suggestion
  description: Mechanism of injury may be, for example: fall, impact (eg: auto accident), weapon (eg: firearm), etc. 

```

File: src/ingest_validation_tools/table-schemas/donor.yaml [hunk 7]
```diff
@@ -49,21 +52,22 @@ fields:
...
+-
+  name: medical_history
+  description:  A record of a patient's background regarding health and the occurrence of disease events of the individual.
+-
+  name: race
+  description:  A geographic ancestral origin category that is assigned to a population group based mainly on physical characteristics that are thought to be distinct and inherent.
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/donor.yaml:64]:
```suggestion
  description:  A grouping of humans based on shared physical characteristics or social/ethnic identity generally viewed as distinct.

```

File: src/ingest_validation_tools/table-schemas/donor.yaml [hunk 8]
```diff
@@ -49,21 +52,22 @@ fields:
...
+  description: How, with what force, and on which part of the body the patient was injured.
+-
+  name: medical_history
+  description:  A record of a patient's background regarding health and the occurrence of disease events of the individual.
+-
+  name: race
+  description:  A geographic ancestral origin category that is assigned to a population group based mainly on physical characteristics that are thought to be distinct and inherent. 
+-
+  name: sex
+  description:  Phenotypic expression of chromosomal makeup that defines a study subject as male, female, or other. Compare to gender.
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/donor.yaml:67]:
```suggestion
  description:  Biological sex at birth: male or female or other.

```

File: src/ingest_validation_tools/table-schemas/donor.yaml [hunk 9]
```diff
@@ -2,31 +2,34 @@
...
-#   description: TODO
-# -
-#   name: cause_of_death
-#   description: TODO
+-
+  name: body_mass_index_value
+  description: An individual's weight in kilograms divided by the square of the height in meters. 
+-
+  name: cause_of_death
+  description:  The circumstance or condition that results in the death of a living being.
```
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/donor.yaml:26]:
```suggestion
  description:  The circumstance or condition that caused death.

```

**Review by cebriggs7135 — CHANGES_REQUESTED** [review]

File: src/ingest_validation_tools/table-schemas/donor.yaml (hunk 5 above)
Comment (mccalluc) [reply src/ingest_validation_tools/table-schemas/donor.yaml:35]: No: The `unit` is different than the `value`.

File: docs/field-descriptions.yaml [hunk 10]
```diff
@@ -28,6 +29,15 @@ assay_type: The specific type of assay being executed.
 bead_barcode_offset: Position(s) in the read at which the bead barcode starts
 bead_barcode_read: Which read file contains the bead barcode
 bead_barcode_size: Length of the bead barcode in base pairs
+blood_type: 'The specific reaction pattern of erythrocytes of an individual to the
+  antisera of one blood group; e.g., the ABO blood group consists of four major blood
+  types: O, A, B, and AB. This classification depends on the presence or absence of
+  two major antigens: A or B. Type O occurs when neither is present and type AB when
+  both are present. The blood type is the genetic phenotype of the individual for
+  one blood group system and may be determined using different antisera available
+  for testing.'
```
[the remark this answers is not in the document]
Comment (mccalluc) [reply docs/field-descriptions.yaml:38]: fixed.

**Pull request merged**